product_form_purchase_link_oca/views/product_template_views.xml

```diff
@@ -0,0 +1,16 @@
+<?xml version="1.0" encoding="utf-8" ?>
+<odoo>
+    <record id="view_product_template_purchase_buttons_from" model="ir.ui.view">
+        <field name="name">product.template.form.view</field>
+        <field name="model">product.template</field>
+        <field name="inherit_id" ref="product_form_purchase_link.view_product_template_purchase_buttons_from"/>
+        <field name="arch" type="xml">
+            <xpath expr="//field[@name='purchased_product_qty']" position="attributes"> 
+                <attribute name="invisible">1</attribute>
+            </xpath>
+            <xpath expr="//button[@class='oe_stat_button']" position="inside"> 
+                <field name="purchased_product" />
```

Comment:
This should fire an error accessing product template form, because this field is already defined for a product variant (`product.product`). And, if access is within product variant form, no change is actually made.

Take a look at original addons:
* Field is already defined for both product template (actually the sum of all field values for its variants) and for the variant (see `purchase` from Odoo S.A.)
* Field is added for both product template and product variant forms (see `product_form_purchase_link` from OCA).

So some stuff is missing:
* Defining new field also for product template.
* Replacing old field by new one in both form views, not only one.